src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -37,59 +309,481 @@ const ActorInit En_Zo_InitVars = {
+    }
+    EnZo_SetAnimation(this);
+
+    Math_SmoothScaleMaxF(&this->actor.velocity.y, this->actor.waterY < 54.0f ? -0.6f : 0.6f, 0.3f, 0.2f);
+    if (this->rippleTimer != 0) {
+        this->rippleTimer--;
+        if ((this->rippleTimer == 3) || (this->rippleTimer == 6)) {
+            EnZo_TreadWater_Ripples(this, 0.2f, 1.0f, 0xC8);
+        }
+    } else {
+        EnZo_TreadWater_Ripples(this, 0.2f, 1.0f, 0xC8);
+        this->rippleTimer = 0xC;
+    }
+
+    if (EnZo_PlayerInProximity(this, globalCtx) != 0) {
+        this->timeToDive = Math_Rand_S16Offset(0x28, 0x28);
+    } else if (DECR(this->timeToDive) == 0) {
+        f32 initialFrame;
+        func_80034EC0(&this->skelAnime, sAnimations, 4);
+        this->canSpeak = 0;
+        this->unk_64C = 1;
+        this->actionFunc = EnZo_Dive;
+        initialFrame = this->skelAnime.initialFrame;
+        this->skelAnime.initialFrame = this->skelAnime.animFrameCount;
+        this->skelAnime.animCurrentFrame = this->skelAnime.animFrameCount;
+        this->skelAnime.animFrameCount = initialFrame;
+        this->skelAnime.animPlaybackSpeed = -1.0f;
+    }
+}
+
+void EnZo_Dive(EnZo* this, GlobalContext* globalCtx) {
+    if (func_800A56C8(&this->skelAnime, this->skelAnime.animFrameCount) != 0) {
+        Audio_PlayActorSound2((Actor*)this, NA_SE_EV_DIVE_WATER);
+        EnZo_SpawnSplashes(this);
+        this->actor.flags &= ~1;
+        this->actor.velocity.y = -4.0f;
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+    }
+
+    if (!(this->skelAnime.animPlaybackSpeed > 0.0f)) {
+        if (this->actor.waterY > 80.0f || this->actor.bgCheckFlags & 1) {
+            Math_SmoothScaleMaxF(&this->actor.velocity.y, -1.0f, 0.4f, 0.6f);
+            Math_SmoothScaleMaxF(&this->alpha, 0.0f, 0.3f, 10.0f);
+        }
+        if ((s16)this->alpha == 0) {
+            func_80034EC0(&this->skelAnime, sAnimations, 2);
+            this->actor.posRot.pos = this->actor.initPosRot.pos;
+            this->alpha = 0.0f;
+            this->actionFunc = EnZo_Submerged;
+        }
+    }
+}
+
+void EnZo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+    ColliderCylinder* collider;
+    Vec3f pos;
+
+    if ((s32)this->alpha != 0) {
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+        EnZo_Blink(this);
+    }
+    Actor_MoveForward((Actor*)this);
+    func_8002E4B4(globalCtx, (Actor*)this, this->collider.dim.radius, this->collider.dim.height * 0.25f, 0.0f, 5);
+    this->actionFunc(this, globalCtx);
+    EnZo_Dialog(this, globalCtx);
+
+    // Spawn air bubbles
+    if (globalCtx->state.frames & 8) {
+        pos = this->actor.posRot.pos;
+
+        pos.y += (Math_Rand_ZeroOne() - 0.5f) * 10.0f + 18.0f;
+        pos.x += (Math_Rand_ZeroOne() - 0.5f) * 28.0f;
+        pos.z += (Math_Rand_ZeroOne() - 0.5f) * 28.0f;
+        EnZo_Bubble(this, &pos);
+    }
+
+    if ((s32)this->alpha != 0) {
+        collider = &this->collider;
```

Comment:
This temp is most likely not needed, but you might have to replace it with a pad and use `thisx` to some extent.